src/libguac/string.c

```diff
@@ -44,6 +45,16 @@
  */
 #define REMAINING(n, length) (((n) < (length)) ? 0 : ((n) - (length)))
 
+size_t guac_itoa(char* restrict dest, int integer) {
+
+    /* Determine size of string. */
+    int str_size = snprintf(dest, 0, "%d", integer);
```

Comment:
This is a signed integer specifier (as opposed to `%i` for unsigned) but neither the type nor the documentation for this function indicate that fact.

Is that intentional?

Reply:
Nah - I should probably be consistent with the return type of the function and just use `size_t`.

Reply:
Ah, nevermind, misunderstood what you were saying. Will switch it over to an unsigned integer.

Reply:
Okay, changed it to "%i" and updated the function documentation and the type to make it `unsigned`.

Reply:
Ugh, sorry I typed that backwards when I commented. `%i` is signed, and `%d` is unsigned. So this is still wrong. My bad.

Reply:
Actually, according to the man page for the `*printf` functions:
```
       d, i   The  int  argument  is  converted  to signed decimal notation.  The precision, if any, gives the minimum number of digits that must appear; if the converted value requires
              fewer digits, it is padded on the left with zeros.  The default precision is 1.  When 0 is printed with an explicit precision 0, the output is empty.

       o, u, x, X
              The unsigned int argument is converted to unsigned octal (o), unsigned decimal (u), or unsigned hexadecimal (x and X) notation.  The letters abcdef are used for x  conver‐
              sions;  the  letters  ABCDEF are used for X conversions.  The precision, if any, gives the minimum number of digits that must appear; if the converted value requires fewer
              digits, it is padded on the left with zeros.  The default precision is 1.  When 0 is printed with an explicit precision 0, the output is empty.
```

So, seems like maybe I should try "u", instead??